src/gui/base/List.ts

```diff
@@ -417,6 +415,10 @@ export class List<T, VH extends ViewHolder<T>> implements ClassComponent<ListAtt
 				row.domElement.style.transform = `translateY(${row.top}px)`
 				row.row.update(item, attrs.state.selectedItems.has(item), attrs.state.inMultiselect)
 			}
+			// Focus the selected row so it can receive keyboard events
```

Comment:
I am concerned that we can't just grab focus here. What if user is in another part of the app but the list get updated?
I think we would need to check if focused element is inside of this list

Reply:
That is a good point. However if we only focus the selected element when the focus is on another item, I suspect this will break tab navigation as it will continually jump to the selected row on tab. I will investigate better places to focus it.

Reply:
I am stumped on finding a way to fix this without refactoring the List's code to allow the `ListModel` to get the row's element from `List.`

Reply:
That wouldn't be desirable.

I am not sure I understand the concern you expressed above. To me the algorithm is "focus the list row *programmatically* only when the focus is already inside the list". This doesn't change anything about manual tab navigation I think? Am I missing something?

Reply:
Ah, I thought programmatically focusing an element affected the tab index. My bad, doing a quick test shows otherwise.

The other issue is that requiring the focus to be inside the list row will change the behaviour in the other layouts. In master, pressing up and down anywhere regardless of focus in the double and three column views will change the selected row. This occurs when I tried only focusing when another row is focused via getting the tag name of the focused element.

Reply:
that's a good thing to think about but I think it still works.

Say you are in three-column layout. You press up/down.
1. If your focus was in the list, it will move to another position, which is good
2. If your focus was not in the list, it will just do the selection, without changing your focus, which is also good

Reply:
The issue's steps do not mention focusing the list for single-column so it would be best not to require it. I have pushed a compromise between the two now.

Reply:
No worries, thanks for taking the time to review it. :slightly_smiling_face: